s-ramp-ide/srampRepositoryBrowser/src/main/java/cz/muni/fi/srampRepositoryBrowser/dialogs/Filter.java

```diff
@@ -0,0 +1,529 @@
+				Job refresh = new RefreshJob("refreshing", Filter.this.parent);
+				refresh.schedule();
+			}
+		});
+		new Label(this, SWT.NONE);
+
+		btnClearAllFilters = new Button(this, SWT.NONE);
+		btnClearAllFilters.setEnabled(true);
+		btnClearAllFilters.setLayoutData(new GridData(SWT.CENTER, SWT.CENTER,
+				false, false, 1, 1));
+		btnClearAllFilters.setText("Clear all filters");
+		btnClearAllFilters.addSelectionListener(new SelectionAdapter() {
+			public void widgetSelected(SelectionEvent event) {
+				Filter.this.parent.setFilter(Filter.this.parent.getManager()
+						.listAllArtifacts());
+				typeT.setText("");
+				createdByT.setText("");
+				lastModifiedByT.setText("");
+				createdFrom = null;
+				createdTo = null;
+				lastModifFrom = null;
+				lastModifTo = null;
+
+				createdFromT.setText("Any");
+				createdToT.setText("Any");
+				lastModifiedFrom.setText("Any");
+				lastModifiedToT.setText("Any");
+
+				Job refresh = new RefreshJob("refreshing", Filter.this.parent);
+				refresh.schedule();
+			}
+		});
+
+	}
+
+	/**
+	 * Find artifact model from artifact type.
+	 * 
+	 * @param type
+	 * @return artifact model
+	 */
+	private String findModel(String type) {
+		switch (type) {
+		// core
+		case "Document":
+		case "XmlDocument":
+			return "core";
+
+			// xsd
+		case "XsdDocument":
+		case "AttributeDeclaration":
+		case "ElementDeclaration":
+		case "SimpleTypeDeclaration":
+		case "ComplexTypeDeclaration":
+			return "xsd";
+
+			// wsdl
+		case "WsdlDocument":
+		case "WsdlService":
+		case "Port":
+		case "WsdlExtension":
+		case "Part":
+		case "Message":
+		case "Fault":
+		case "PortType":
+		case "Operation":
+		case "OperationInput":
+		case "OperationOutput":
+		case "Binding":
+		case "BindingOperation":
+		case "BindingOperationInput":
+		case "BindingOperationOutput":
+		case "BindingOperationFault":
+			return "wsdl";
+
+			// policy
+		case "PolicyDocument":
+		case "PolicyExpression":
+		case "PolicyAttachment":
+			return "policy";
+
+			// soa
+		case "HumanActor":
+		case "Choreography":
+		case "ChoreographyProcess":
+		case "Collaboration":
+		case "CollaborationProcess":
+		case "Composition":
+		case "Effect":
+		case "Element":
+		case "Event":
+		case "InformationType":
+		case "Orchestration":
+		case "OrchestrationProcess":
+		case "Policy":
+		case "PolicySubject":
+		case "Process":
+		case "Service":
+		case "ServiceContract":
+		case "ServiceComposition":
+		case "ServiceInterface":
+		case "System":
+		case "Task":
+			return "soa";
+
+			// serviceImplementation
+		case "Organization":
+		case "ServiceEndpoint":
+		case "ServiceInstance":
+		case "ServiceOperation":
+			return "serviceImplementation";
+
+		default:
+			return "ext";
+		}
+	}
+
+	/**
+	 * Method set query by typeT field (type of artifact), createdByT,
+	 * lastModifiedByT, createdFrom, createdTo, lastModifFrom and lastModifTo
+	 * field.
+	 */
+	private SrampClientQuery setFilter() {
```

Comment:
FYI, this section may be *really* helpful for something we're working on within the UI.  I may end up pulling it into a common utility.